File: ir/dict_ir.py
from common.span import SourceSpan
from ir.expr_ir import ExprIR


class DictIR(ExprIR):
    def __init__(
        self,
        keys: list[ExprIR | None],
        values: list[ExprIR],
        span: SourceSpan,
    ):
        super().__init__(span=span)
        self.keys = keys
        self.values = values

File: rust/src/ir/nodes/dict_ir.rs
use crate::ir::expr_ir::ExprIR;
use crate::ir::span_ir::SourceSpan;

#[derive(Debug, Clone)]
pub struct DictIR {
    pub elements: Vec<DictEntryIR>,
    pub span: Option<SourceSpan>,
}

#[derive(Debug, Clone)]
pub struct DictEntryIR {
    pub key: Option<ExprIR>, // {**something} is allowed
    pub value: ExprIR,
    pub span: Option<SourceSpan>,
}
